src/frontend/planner_test/tests/testdata/agg.yaml

```diff
@@ -696,6 +696,27 @@
         └─StreamExchange { dist: Single }
           └─StreamStatelessLocalSimpleAgg { aggs: [count, sum(t.v2) filter((t.v2 < 5:Int32))] }
             └─StreamTableScan { table: t, columns: [t.v2, t._row_id], pk: [t._row_id], dist: UpstreamHashShard(t._row_id) }
+- name: force two phase aggregation
+  sql: |
+    SET QUERY_MODE TO DISTRIBUTED;
+    SET RW_FORCE_TWO_PHASE_AGG=true;
+    create table t(v1 int, v2 smallint, v3 varchar);
+    select min(v3), sum(v1) from t group by v1, v3, v2;
+  batch_plan: |
+    BatchExchange { order: [], dist: Single }
+    └─BatchProject { exprs: [min(min(t.v3)), sum(sum(t.v1))] }
+      └─BatchHashAgg { group_key: [t.v1, t.v3, t.v2], aggs: [min(min(t.v3)), sum(sum(t.v1))] }
+        └─BatchHashAgg { group_key: [t.v1, t.v3, t.v2], aggs: [min(t.v3), sum(t.v1)] }
+          └─BatchExchange { order: [], dist: HashShard(t.v1, t.v2, t.v3) }
+            └─BatchScan { table: t, columns: [t.v1, t.v2, t.v3], distribution: SomeShard }
+  stream_plan: |
+    StreamMaterialize { columns: [min, sum, t.v1(hidden), t.v3(hidden), t.v2(hidden)], pk_columns: [t.v1, t.v3, t.v2] }
+    └─StreamProject { exprs: [min(min(t.v3)), sum(sum(t.v1)), t.v1, t.v3, t.v2] }
+      └─StreamHashAgg { group_key: [t.v1, t.v3, t.v2], aggs: [count, min(min(t.v3)), sum(sum(t.v1))] }
+        └─StreamExchange { dist: HashShard(t.v1, t.v3, t.v2) }
```

Comment:
Seems like exchange is still needed since we distribute on `Vnode(t._row_id)`. Wonder if this can be improved.